### PR TITLE
[FSK-88]: List Of Families - Implement The List

### PR DESCRIPTION
**Pull request by AmirabbasJ**
# What changed

- list of families is implemented 
- refactored DashboardCard
- made `CreateFamilyButton` and `CreateFamilyModa` use a shared context for mutating and using modalState
- changed badge colors
- add `CreateFamilyButton` to the empty state of families and the same for `CreateProjectButton`
- add component tests and storybook for `familyList`

### CODE REVIEW

**Comment by erfan-goodarzi** [comment]
@AmirabbasJ `CreateFamilyModal` [storybook](https://6315924b26061e4ff69d1b5b-lvnerkgnli.chromatic.com/?path=/story/components-createfamily-familyform-createfamilymodal--default) is broken.

**Review by ASafaeirad — APPROVED** [review]

File: @types/route.d.ts [hunk 1]
```diff
@@ -8,5 +8,7 @@ type AppRoute =
   | '/auth/login'
   | '/dashboard'
   | '/dashboard/families'
+  | '/dashboard/families/:id'
   | '/dashboard/families/family-detail'
+  | '/dashboard/family-detail'
```
Comment (ASafaeirad) [inline @types/route.d.ts:13]: Why these two? `/families/:id` is enough

File: src/components/CreateFamily/FamilyForm/CreateFamilyModal.cy.tsx [hunk 2]
```diff
@@ -1,11 +1,19 @@
...
+import { CreateFamilyModalStateCtx } from '../../../contexts';
 import { createFamilyFormIds } from './CreateFamilyForm';
 import { CreateFamilyModal, createFamilyModalId } from './CreateFamilyModal';
 
 describe('Create Family Modal', () => {
   beforeEach(() => {
-    cy.mount(<CreateFamilyModal opened onClose={noop} />);
+    cy.mount(
+      <CreateFamilyModalStateCtx.Provider
+        // eslint-disable-next-line react/jsx-no-constructed-context-values
```
Comment (ASafaeirad) [inline src/components/CreateFamily/FamilyForm/CreateFamilyModal.cy.tsx:11]: Let's disable this rule globally for `.cy.ts` files.

File: src/components/CreateFamily/FamilyForm/CreateFamilyModal.stories.tsx [hunk 3]
```diff
@@ -14,8 +15,15 @@ export default {
   component: CreateFamilyModal,
 } as Meta;
 
-const Template: Story<CreateFamilyModalProps> = args => {
-  return <CreateFamilyModal {...args} />;
+const Template: Story<typeof CreateFamilyModal> = () => {
+  return (
+    <CreateFamilyModalStateCtx.Provider
+      // eslint-disable-next-line react/jsx-no-constructed-context-values
```
Comment (ASafaeirad) [inline src/components/CreateFamily/FamilyForm/CreateFamilyModal.stories.tsx:21]: Let's disable this rule globally for `.stories` files.

File: src/main.tsx [hunk 4]
```diff
@@ -14,9 +15,11 @@ root.render(
     <ApolloProvider>
       <ThemeProvider>
         <NotificationsProvider limit={3}>
-          <AuthProvider fallback={<FullPageLoader />}>
-            <Routes />
-          </AuthProvider>
+          <ContextProvider>
```
Comment (ASafaeirad) [inline src/main.tsx:18]: I wouldn't say I like this idea. It kind of means contexts are global states.

**Review by ASafaeirad — CHANGES_REQUESTED** [review]

File: @types/route.d.ts (hunk 1 above)
Comment (AmirabbasJ) [reply @types/route.d.ts:13]: they were they from the start so I leaved them there for now
but yeah you're right it shouldn't break anything

File: src/main.tsx (hunk 4 above)
Comment (AmirabbasJ) [reply src/main.tsx:18]: agree

File: src/contexts/CreateFamilyModalStateCtx.tsx [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+  const [isModalOpen, { setFalse: closeModal, setTrue: openModal }] =
+    useBoolean(false);
+
+  const ctx: CreateFamilyModalStateCtx = useMemo(
+    () => ({
+      isModalOpen,
+      closeModal,
+      openModal,
+    }),
+    [isModalOpen, closeModal, openModal],
```
Comment (ASafaeirad) [inline src/contexts/CreateFamilyModalStateCtx.tsx:25]: Let's use https://mantine.dev/others/modals/

**Review by ASafaeirad — CHANGES_REQUESTED** [review]

**Review by ASafaeirad — APPROVED** [review]

**Pull request merged**